catkit/hardware/iris_ao/iris_ao_controller.py

```diff
@@ -1,109 +1,167 @@
-                   disable_hardware,
-                   path_to_dm_exe,
-                   filename_ptt_dm,
-                   path_to_custom_mirror_files=None):
+                   mcf_filepath,
+                   dcf_filepath,
+                   dll_filepath,
+                   disable_hardware=False):
         """
         Initialize dm manufacturer specific object - this does not, nor should it, open a
         connection.
 
         :param mirror_serial: string, The mirror serial number. This corresponds to a .mcf file that MUST include the
                               driver serial number under "Smart Driver". See README.
         :param driver_serial: string, The driver serial number. This corresponds to a .dcf file. See README.
-        :param disable_hardware: bool, If False, will run on hardware (always used on JOST this way). If True,
-                                 probably (!) just runs the GUI? We never used it with True, so not sure.
-        :param path_to_dm_exe: string, The path to the local directory that houses the DM_Control.exe file.
-        :param filename_ptt_dm: string, Full path including filename of the ini file that provides the PTT values to be
-                                loaded onto the hardware, e.g. ".../ConfigPTT.ini".
-        :param path_to_custom_mirror_files: string, Full path to mirror .ini files.
+        :param mcf_filepath: string, Full path to .mcf file.
+        :param dcf_filepath: string, Full path to .dcf file.
+        :param dll_filepath: string, Full path to IrisAO.Devices.dll (x64).
+        :param disable_hardware: bool, False := use hardware, True := all hardware APIs are NOOPs.
         """
 
-        self.log.info("Opening IrisAO connection")
-        # Create class attributes for storing an individual command.
-        self.command_object = None
+        self.dll_filepath = dll_filepath
+        self.instrument_lib = self.instrument_lib(self.dll_filepath)
 
+        self.command_object = None
         self.mirror_serial = mirror_serial
         self.driver_serial = driver_serial
-
-        # For the suprocess call
         self.disable_hardware = disable_hardware
-        self.path_to_dm_exe = path_to_dm_exe
-        self.full_path_dm_exe = os.path.join(path_to_dm_exe, 'DM_Control.exe')
-        self.path_to_custom_mirror_files = path_to_custom_mirror_files
-
-        # Where to write ConfigPTT.ini file that gets read by the C++ code
-        self.filename_ptt_dm = filename_ptt_dm
-
-    def send_data(self, data):
+        self.mcf_filepath = mcf_filepath
+        self.dcf_filepath = dcf_filepath
+
+        # Check config files exist.
+        self.validate_config_files()
+
+        # The IrisAO driver expects the .dcf and .mcf config files in the working dir.
+        # Neither adding them to PATH nor PYTHONPATH worked.
+        # This is a workaround.
+        # These files remain copied until their refs go out of scope, i.e., when this class instance drops out of scope.
```

Comment:
This is an old comment from when we tried using the vendor's python API.

It's more likely that these need to go in the same dir as the dll.